RHIDP-3440-install CX: Add plugin install content (#547)
redhat-developer/red-hat-developers-documentation-rhdh: modules/dynamic-plugins/con-rhdh-plugins.adoc

[id="con-rhdh-plugins"]

> = Plugins in {product}

Author: This content was moved to the About title...I'll review your suggestions in this PR, once it's merged it will then cover #555

The {product} ({product-very-short}) application offers a unified platform with various plugins. Using the plugin ecosystem within the {product-very-short} application, you can access any kind of development infrastructure or software development tool.

Plugins are modular extensions for {product-very-short} that extend functionality, streamline development workflows, and improve the developer experience. You can add and configure plugins in {product-very-short} to access various software development tools.

Each plugin is treated as a self-contained application and can include almost any type of content. Plugins use a common set of platform APIs and reusable UI components. Plugins can get data from external sources using APIs or by depending on external modules to do the work.

{product-very-short} provides both static plugins and dynamic plugins which extend {product-very-short} functionality. Static plugins are built into the core of the {product-very-short} application, dynamic plugins are sideloaded into your Developer Hub instance without having to recompile your code or rebuild the container. 

> To install or update a static plugin you must update your {product-very-short} application source code and rebuild the application and container image.

Author: AFAIU Static plugins are built-in to the application source code, so I'm not sure how we would approach that but I agree we should review this after 1.3 to see if it's something we can document. I'll take note.

Modularity and extensibility::
Dynamic plugins allow teams to add or modify features without altering the core {product-very-short} application. This modular approach makes it easier to extend functionality as needs evolve.

Customization::
Different teams or projects might have unique requirements. Dynamic plugins enable users to tailor {product-very-short} to fit specific workflows and use cases, enhancing the overall user experience.

Reduced maintenance and update overhead:: 
With dynamic plugins, updates or new features can be deployed independently of the main {product-very-short} codebase. This reduces the risks and efforts associated with maintaining and updating the platform.

Faster iteration:: 
Developers can create and test new features more rapidly as plugins. This encourages experimentation and innovation, allowing teams to quickly iterate based on feedback.

Improved collaboration::
Organizations can share plugins across teams or even externally. This sharing can foster collaboration and reduce duplication of effort, as well as help establish best practices across an organization.

Scalability:: 

> As organizations grow, their needs become more complex. Dynamic plugins enable {product-very-short} to scale alongside these needs, accommodating an increasing number of users and services.

Reviewer: ```suggestion
As organizations grow, their needs become complex. Dynamic plugins enable {product-very-short} to scale alongside such complex needs, accommodating an increasing number of users and services.
```

Ecosystem growth:: 

> Encouraging the development of plugins can lead to a vibrant ecosystem around {product-very-short}. This community can contribute plugins that address a wide range of needs, further enriching the platform.

Reviewer: ```suggestion
Fostering the development of plugins can create a dynamic ecosystem around {product-very-short}. This community can contribute to plugins that cater to different needs, thereby enhancing the platform.
```

Security and compliance:: 

> Plugins can be developed with specific security and compliance requirements in mind, ensuring that {product-very-short} installations meet the necessary standards without compromising the core application.

Reviewer: ```suggestion
You can develop plugins with specific security and compliance requirements in mind, ensuring that {product-very-short} installations meet the necessary standards without compromising the core application.
```

Overall, the use of dynamic plugins in {product-very-short} promotes a flexible, adaptable, and sustainable approach to managing and scaling development infrastructure.

> == Comparing dynamic plugins to static plugins

Author: This entire section belongs in the About/Introduction to plugins guide...not here at all.
